assert/assertions.go

```diff
@@ -909,3 +910,37 @@ func NotZero(t TestingT, i interface{}, msgAndArgs ...interface{}) bool {
 	}
 	return true
 }
+
+// JSONEq asserts that two JSON strings are equivalent.
+//
+//  assert.JSONEq(t, `{"hello": "world", "foo": "bar"}`, `{"foo": "bar", "hello": "world"}`)
+//
+// Returns whether the assertion was successful (true) or not (false).
+func JSONEq(t TestingT, expected string, actual string, msgAndArgs ...interface{}) bool {
+	expectSlice := false
+	expectedJSONAsMap := make(map[string]interface{})
+	expectedJSONAsSlice := make([]interface{}, 0, 0)
```

Comment:
you can `Unmarshal` into an `interface{}` rather than doing the two attempts:

```go
package main

import "fmt"
import "encoding/json"

func main() {
	var x interface{}
	json.Unmarshal([]byte("{\"foo\": \"bar\"}"), &x)
	fmt.Printf("%#v\n", x)
	json.Unmarshal([]byte("[{\"foo\": \"bar\"}]"), &x)
	fmt.Printf("%#v\n", x)
}
```

```
map[string]interface {}{"foo":"bar"}
[]interface {}{map[string]interface {}{"foo":"bar"}}
```